src/LEGO.AsyncAPI.Bindings/Sqs/Principal.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) The LEGO Group. All rights reserved.
+
+namespace LEGO.AsyncAPI.Bindings.Sqs;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using LEGO.AsyncAPI.Models.Interfaces;
+using LEGO.AsyncAPI.Readers.ParseNodes;
+
+public class Principal : IAsyncApiElement
+{
+    public Principal(KeyValuePair<string, StringOrStringList> value)
```

Comment:
instead of this and the other ctor.
Keep what you had, and instead make a `public implicit operator Principal(PrincipalStar principal) => new Principal(principal)` and `public implicit operator Principal(PrincipalObject principal) => new Principal(principal)` in each of trhe 2 `PrincipalValue` classes.

This will allow us to implicitly convert from those object types into Principal. so we can use them directly instead of going through `Principal` itself..

Reply:
but looking at all of this.
you could also set the `Principal` of `Statement.cs` to be `IPrincipalValue` and move the PrincipalParse stuff into that class.
And get rid of Principal.cs altogether.